### ISSUE TITLE
Issues with recognizing 1s from 4s

### ISSUE THREAD

**Issue by mahmoudaljan** [aprettyloner/ocr-autograder#1]
Hey, thank you for your great work ! i have been trying to use your project and regularly had issues with confusing 1s for 4s, can you help ?

![cropped_image](https://user-images.githubusercontent.com/26923780/79637727-aa839f00-8179-11ea-8547-e18068765992.jpg)

![cropped_image](https://user-images.githubusercontent.com/26923780/79637737-ba02e800-8179-11ea-8a48-bf37655d1f15.jpg)

**Comment by aprettyloner** [comment]
Thanks for the feedback! 

**TL;DR** 
I would test out previous models, see these in the [static directory](https://github.com/aprettyloner/ocr-autograder/tree/master/my_app/static). The model is controlled with the following line in [main.py](https://github.com/aprettyloner/ocr-autograder/blob/master/my_app/main.py).

`model='static/<try_other_pkl_or_h5_here>'`

 In particular, I might test `mnist_hasyv2_master_20epochs_batch64__ALLDATA_201911081573211546.h5`.

Can you please provide me with the output for the two images you provided? 4430 and 4830?

**Longer answer** 
This issue is related to the training set. Some of my previous models correctly detected 1s that are written the way you provided above. However, this introduced some confusion with 7 detection, especially in my use case (USA). What I did to fix this was I omitted 1s and 9s from the Kensanata training set for the final model. As you can see below, these digits are written quite differently than the MNIST set, which I believe is from the USA. This causes some conflicts in the model.

For details on my modeling process and evaluation, [please see here](https://github.com/aprettyloner/ocr-autograder).


![Screen Shot 2020-04-18 at 7 04 12 AM](https://user-images.githubusercontent.com/22802852/79639851-0a128800-8143-11ea-911f-1afbb2e160e8.png)


For more context, digits from the Kensanata set look like this:
![image](https://user-images.githubusercontent.com/22802852/79639349-3a0c5c00-8140-11ea-96c3-456dfba1d63e.png)

**Comment by mahmoudaljan** [comment]
Hello ! I just tried the model you recommended and it worked like a charm ! I will continue doing further tests on more data to check. so far so good !

**Comment by aprettyloner** [comment]
Great to hear! Thanks again for your feedback. 

I'll close this issue for now, but please keep me updated with data checks!

**Comment by mahmoudaljan** [comment]
Here are the results for the FINAL___tf_model___201911191574216513.h5 model
![1](https://user-images.githubusercontent.com/26923780/79640953-8da49700-818c-11ea-88ee-76a9a13daf44.JPG)
![2](https://user-images.githubusercontent.com/26923780/79640954-8e3d2d80-818c-11ea-8790-bfc0cf51cc6c.JPG)

**Comment by mahmoudaljan** [comment]
I am also curious to see how eventually google's AutoML would manage this problem automatically even when mixing the datasets, 🤔🤔🤔

**Comment by aprettyloner** [comment]
Good question. If you work it out with AutoML let me know.

Also, could you post the final results for the working model (like above)?

**Comment by mahmoudaljan** [comment]
Hello again,
Here are the results, i also included two more examples, one of a printed set of digits and one more where pre-ocr processing was a bit spotty.

![Capture](https://user-images.githubusercontent.com/26923780/79687159-36aaca80-823d-11ea-9c6e-3003c863dca0.JPG)
![Capture2](https://user-images.githubusercontent.com/26923780/79687160-37dbf780-823d-11ea-9c35-7878cbf7754f.JPG)
![Capture3](https://user-images.githubusercontent.com/26923780/79687161-38748e00-823d-11ea-9cea-536b8d4c0be2.JPG)
![Capture4](https://user-images.githubusercontent.com/26923780/79687162-390d2480-823d-11ea-9ff2-8d7bca77d6b5.JPG)

**Comment by aprettyloner** [comment]
Good to note. Thanks @mahmoudaljan
